api/src/organizational-entities/infrastructure/repositories/certification-center-for-admin.repository.js

```diff
@@ -22,7 +23,28 @@ const update = async function (certificationCenter) {
     .where({ id: certificationCenter.id });
 };
 
-export { save, update };
+/**
+ * @type {function}
+ * @param {Object} params
+ * @param {string|number} params.certificationCenterId
+ * @param {string|number} params.archivedBy
+ * @param {date} params.archiveDate
+ */
+const archive = async function ({ certificationCenterId, archivedBy, archiveDate }) {
+  const knexConn = DomainTransaction.getConnection();
+  const certificationCenter = await knexConn('certification-centers').where({ id: certificationCenterId }).first();
+  if (!certificationCenter) {
+    throw new CertificationCenterNotFoundError();
+  }
+
+  if (certificationCenter.archivedBy === null) {
+    await knexConn('certification-centers')
+      .where({ id: certificationCenterId })
+      .update({ archivedBy, archivedAt: archiveDate });
+  }
```

Comment:
Il serait plus clair de coder cette partie avec une logique _exit-early_ : 

```suggestion
  if (certificationCenter.archivedBy) {
    return;
  }
  
  await knexConn('certification-centers')
    .where({ id: certificationCenterId })
    .update({ archivedBy, archivedAt: archiveDate });
```